### PR TITLE
React assesment part-1

### PR DESCRIPTION
**Pull request by aryan-249**

### CODE REVIEW

File: src/homePage.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import React, { Component } from "react";
+import {Link} from "react-router-dom";
+import "./styles.css";
+import Todolist from "./todolist";
+class App extends Component {
+  // React States  
+    state = {
```
Comment (CryptoKage2306) [inline src/homePage.js]: Why there is no constructor

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import RegisterForm from "./registerPage";
+import './index.css';
+import App from './homePage.js';
+import Todolist from './todolist'
+
+import reportWebVitals from './reportWebVitals';
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  
+      <div>
```
Comment (CryptoKage2306) [inline src/index.js]: No Indentation?

any specific reason for not doing so

File: src/listentry.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import react from 'react';
+const Row = (input)=>{
+   return (
+    <div>
+        {input.value}
+         <input type="checkbox"></input>
```
Comment (CryptoKage2306) [inline src/listentry.js]: Wrong indentation?

can you please find out the correct indentation?

File: src/registerPage.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import "./styles.css"
+
+class RegisterForm extends Component {
+  // react States
+    state ={
```
Comment (CryptoKage2306) [inline src/registerPage.js]: No Constructor?

File: src/todolist.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import React ,{Component} from 'react'
+import Row from "./listentry"
+
+class Todolist extends Component {
+    constructor(props){
```
Comment (CryptoKage2306) [inline src/todolist.js]: why there is weird indentation?

File: src/todolist.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import React ,{Component} from 'react'
```
Comment (CryptoKage2306) [inline src/todolist.js]: Indentation?

**Review by CryptoKage2306 — COMMENTED** [review]
I havent checked your code yet.


fix indentation first thing in Monday

and let me know

File: src/homePage.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
+import React, { Component } from "react";
+import {Link} from "react-router-dom";
```
Comment (CryptoKage2306) [inline src/homePage.js]: give space between 
```
import { Link } from "react-router-dom";
```

File: src/homePage.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+render(){ 
+  return (
+    <div className="App">
+      
+      {this.state.isloggedin?
+      <div>Logged in Successfully
+        <Link to='/todolist' Component={<Todolist List={this.state.currentUser.taskList}/>}>Get to do list</Link>
+      </div>
+
+      :(<div><h1> Login Form</h1>
```
Comment (CryptoKage2306) [inline src/homePage.js]: Give nextline break between `<div> <h1>` and space between `( <div> `

File: src/index.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import { BrowserRouter as Router, Routes, Route, Link } from "react-router-dom";
+import RegisterForm from "./registerPage";
+import './index.css';
+import App from './homePage.js';
+import Todolist from './todolist'
+
+import reportWebVitals from './reportWebVitals';
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+
```
Comment (CryptoKage2306) [inline src/index.js]: Why unnecessary space

File: src/homePage.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+        },
+        isloggedIn:false,
+      };
+  };
+
+  // User Authentication Function
+  handleSubmit = (event) => {
+
+    // Prevent page reload
+    event.preventDefault();
```
Comment (CryptoKage2306) [inline src/homePage.js:25]: what happen if we remove the event.preventDefault()

File: src/homePage.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  };
+};
+
+render() {
+  return (
+    <div className = "App">
+      <Routes>
+        <Route exact path = "/registerPage" element={<RegisterForm/>}></Route>
+        <Route exact path = '/todolist' element={ <Todolist email={this.state.currentUser.email}  isLoggedIn = {this.state.isloggedIn} todolist = {this.state.currentUser.taskList} password = {this.state.currentUser.password}/>}></Route>
+        <Route exact path='/' element={this.state.isloggedIn ? (<Link to='/todolist'>Todolist</Link>):(<div>
```
Comment (CryptoKage2306) [inline src/homePage.js]: Can you explain the code from L73 to L83

File: src/homePage.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+    // Username not found
+    alert("User Not found")
+  };
+};
+
+render() {
+  return (
+    <div className = "App">
+      <Routes>
+        <Route exact path = "/registerPage" element={<RegisterForm/>}></Route>
```
Comment (CryptoKage2306) [inline src/homePage.js]: why Route is not self closing tag

File: src/registerPage.js [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+        localStorage.setItem(email.value,JSON.stringify({password:password.value,tasklist:[]}));
+        alert(" User Registered successfully");
+    };
+  };
+
+  render () {
+    return (
+      <div className="App">
+        {this.state.isSubmitted
+        ?<div>Registered Successfully</div>
```
Comment (CryptoKage2306) [inline src/registerPage.js]: break a line whenever you are using ternary with div element

File: src/todolist.js [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+class Todolist extends Component {
+  // Initiating Object Constructor
+  constructor (props) {
+	  super(props);
+	  this.state = {
+		  count : props.todolist.length,
+		  value : '',
+	    todolist : props.todolist,
+	    email : props.email,
+      password : props.password
```
Comment (CryptoKage2306) [inline src/todolist.js]: Ye kaisa indentation hai.


Snake game khel rhe ho code ke sath

File: src/todolist.js [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+		<div className="App">
+
+		{this.props.isLoggedIn?
+		<>
+		<h1>TodoList</h1>
+		 <ul>{mylist}</ul>
+		 <input onChange = {this.handleChange}></input>
+		 <button onClick = { () => this.onAddTask( this.state.value )}> Add Task </button>
+	 </>
+	 :<h1>Please Log in</h1>
```
Comment (CryptoKage2306) [inline src/todolist.js]: give a line break whenever you are using ternary op with component

**Review by CryptoKage2306 — COMMENTED** [review]
- Indentation is improved from the last time.
- Ternary op indentation is bad
- refer to airbnb guide for better coding std
- I saw a login functionaliity but no logout functionality

File: src/listentry.js [hunk 16]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Row = (input)=>{
+   return (
+    <div>
```
Comment (abbasali1994) [inline src/listentry.js]: if not in use then remove this file

File: src/homePage.js (hunk 10 above)
Comment (aryan-249) [reply src/homePage.js:25]: Default actions to given component then cannot be prevented.

File: src/listentry.js (hunk 3 above)
Comment (aryan-249) [reply src/listentry.js]: yes i have gone through the guide and find out we use 2 space indentation for each child element.

**Review by abbasali1994 — APPROVED** [review]

**Pull request merged**